Library/Homebrew/formulary.rb

```diff
@@ -255,10 +255,12 @@ def self.load_formula_from_api(name, flags:)
       end
 
       resource "ruby-source" do
-        url "https://raw.githubusercontent.com/Homebrew/homebrew-core/#{json_formula["tap_git_head"]}/Formula/#{name}.rb"
-        if (ruby_source_sha256 = json_formula.dig("ruby_source_checksum", "sha256")).present?
-          sha256 ruby_source_sha256
-        end
+        tap_git_head = json_formula.fetch("tap_git_head", "HEAD")
```

Comment:
Doesn't need to be this PR but: this PR or one following could be a good opportunity to unify these e.g. as described in https://github.com/Homebrew/brew/issues/14746

Reply:
Probably not this PR, I don't yet have the whole picture in mind since I haven't touched API code before. Reopened that issue since it was closed as stale.